### PR TITLE
Use directives to allow for unit testing

### PR DESCRIPTION
**Pull request by kshepard**

### CODE REVIEW

File: schema_editor/app/scripts/views/recordtype/module.js [hunk 1]
```diff
@@ -11,44 +11,33 @@
...
-            controller: 'RTEditController',
-            controllerAs: 'rt'
+            template: '<ase-rt-edit></ase-rt-edit>'
         });
         $stateProvider.state('rt.detail', {
-            url: '/detail/:uuid',
-            templateUrl: 'scripts/views/recordtype/detail-partial.html',
-            controller: 'RTDetailController',
-            controllerAs: 'rtDetail'
+            url: '/edit/:uuid',
```
Comment (CloudNiner) [inline schema_editor/app/scripts/views/recordtype/module.js]: Should still be `/detail/:uuid`

File: schema_editor/app/scripts/views/recordtype/detail-directive.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+(function () {
+    'use strict';
+
+    /* ngInject */
+    function RTDetail() {
+        var module = {
+            restrict: 'E',
+            templateUrl: 'scripts/views/recordtype/detail-partial.html',
+            scope: {
+                filters: '='
```
Comment (CloudNiner) [inline schema_editor/app/scripts/views/recordtype/detail-directive.js]: What is this filters property on all the scopes?

File: schema_editor/app/scripts/views/recordtype/module.js (hunk 1 above)
Comment (kshepard) [inline schema_editor/app/scripts/views/recordtype/module.js]: Good catch.

File: schema_editor/app/scripts/views/recordtype/detail-directive.js (hunk 2 above)
Comment (kshepard) [inline schema_editor/app/scripts/views/recordtype/detail-directive.js]: Bad copy/paste. I've removed them and updated the PR.

**Comment by CloudNiner** [comment]
+1

**Pull request merged**